lib/publiccloud/gce.pm

```diff
@@ -260,6 +260,27 @@ sub stop_instance {
     die("Failed to stop instance $instance_id") unless ($attempts > 0);
 }
 
+sub suspend_instance {
+    my ($self, $instance) = @_;
+    my $instance_id = $instance->instance_id();
+
+    die("Cannot suspend instance which is not running.") if ($self->get_state_from_instance($instance) ne 'RUNNING');
+    assert_script_run("gcloud compute instances suspend $instance_id", timeout => 300);
+    die("Failed to suspend instance") if ($self->get_state_from_instance($instance) ne 'SUSPENDED');
```

Comment:
Probably not needed, but don't we also want to perform multiple attempts here, as we do for EC2?

(also on the following occurrences)

Reply:
Did so.